### ISSUE TITLE
Bad Indentation with `match ... end`

### ISSUE THREAD

**Issue by permui** [rocq-prover/vsrocq#248]
Expected Behavior: The `end` should be aligned with the pairing `match` after input.
Actual Behavior: `end` goes to the outmost indentation.
Illustration: 
![2021-08-24-11-16-03](https://user-images.githubusercontent.com/34205956/130550639-4b898cf5-cd72-4213-bf22-b8b817d8ee2c.gif)
This is quite annoying and I could not find any settings related to this.
Hoping a fix. Thanks very much.

**Comment by rtetley** [comment]
Related to #247 and #158

**Comment by shilangyu** [comment]
Without a full blown formatter, I think what was done in #1015 is the best that can be done regarding this issue.

I had a look at what rust-analyzer does (notoriously well-developed LSP for Rust), it also does not handle this correctly.

**Comment by Durbatuluk1701** [comment]
> Without a full blown formatter, I think what was done in #1015 is the best that can be done regarding this issue.

I tend to agree, I think #158 is probably the optimal tracking issue (for a hopefully eventual formatter)

**Comment by shilangyu** [comment]
@gares I am closing as I believe we have implemented the best version possible. To improve it we would need a full code formatter.
